lib/puppetserver/ca/generate_action.rb

```diff
@@ -206,11 +217,40 @@ def sign_intermediate(ca_key, ca_cert, csr, valid_until, signing_digest)
           extension = ef.create_extension(*ext)
           cert.add_extension(extension)
         end
+
+        if !subject_alt_names.empty?
+          alt_names_ext = ef.create_extension("subjectAltName", subject_alt_names, false)
+          cert.add_extension(alt_names_ext)
+        end
+
         cert.sign(ca_key, signing_digest)
 
         cert
       end
 
+      def choose_alt_names(cli_alt_names, settings_alt_names)
```

Comment:
This would be a good thing to have docs on, I was unsure of the inputs and outputs on the first pass through.